File: version_java/src/main/java/com/example/version_java/ui/register/RegisterActivity.java
package com.example.version_java.ui.register;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.example.version_java.R;
import com.example.version_java.base.BaseActivity;
import com.example.version_java.databinding.ActivityRegisterBinding;
import com.example.version_java.ui.home.HomeActivity;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class RegisterActivity extends BaseActivity {

    private RegisterViewModel registerViewModel;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initUi();
        initViewModel();
    }

    private void initUi() {
        ((ActivityRegisterBinding) getBinding()).inputPassRegisterOk.setOnEditorActionListener((v, actionId, event) -> {
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                registerViewModel.register();
                return true;
            } else {
                return false;
            }
        });
    }

    private void initViewModel() {
        registerViewModel = new ViewModelProvider(this).get(RegisterViewModel.class);
        ((ActivityRegisterBinding) getBinding()).setViewModel(registerViewModel);

        registerViewModel.getViewStateLiveData().observe(RegisterActivity.this, viewState -> {
            if (viewState instanceof RegisterViewState) {
                onChangedViewState((RegisterViewState) viewState);
            }
        });
    }

    private void onChangedViewState(RegisterViewState viewState) {
        if (viewState instanceof RegisterViewState.RouteHome) {
            Intent intent = new Intent(RegisterActivity.this, HomeActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        } else if (viewState instanceof RegisterViewState.Error) {
            Toast.makeText(this, ((RegisterViewState.Error) viewState).getMessage(), Toast.LENGTH_SHORT).show();
        } else if (viewState instanceof RegisterViewState.EnableInput) {
            ((ActivityRegisterBinding) getBinding()).inputEmailRegister.setEnabled(((RegisterViewState.EnableInput) viewState).isEnable());
            ((ActivityRegisterBinding) getBinding()).inputPassRegister.setEnabled(((RegisterViewState.EnableInput) viewState).isEnable());
            ((ActivityRegisterBinding) getBinding()).inputPassRegisterOk.setEnabled(((RegisterViewState.EnableInput) viewState).isEnable());
        } else if (viewState instanceof RegisterViewState.ShowProgress) {
            ((ActivityRegisterBinding) getBinding()).progressbar.bringToFront();
            ((ActivityRegisterBinding) getBinding()).progressbar.setVisibility(View.VISIBLE);
        } else if (viewState instanceof RegisterViewState.HideProgress) {
            ((ActivityRegisterBinding) getBinding()).progressbar.setVisibility(View.GONE);
        }
    }


    public RegisterActivity() {
        super(R.layout.activity_register);
        this.setBinding((ActivityRegisterBinding) this.getBinding());
    }
}

File: app/src/main/java/com/example/crimenotification/ui/register/RegisterActivity.kt
package com.example.crimenotification.ui.register

import android.content.Intent
import android.os.Bundle
import android.view.inputmethod.EditorInfo
import androidx.activity.viewModels
import androidx.core.view.isVisible
import com.example.crimenotification.R
import com.example.crimenotification.base.BaseActivity
import com.example.crimenotification.databinding.ActivityRegisterBinding
import com.example.crimenotification.ext.showToast
import com.example.crimenotification.ui.home.HomeActivity
import com.example.crimenotification.ui.map.MapViewState
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class RegisterActivity : BaseActivity<ActivityRegisterBinding>(R.layout.activity_register) {

    private val registerViewModel by viewModels<RegisterViewModel>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initUi()
        initViewModel()
    }

    private fun initUi() {
        binding.inputPassRegisterOk.setOnEditorActionListener { _, actionId, _ ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                registerViewModel.register()
                true
            } else {
                false
            }
        }
    }

    private fun initViewModel() {
        binding.viewModel = registerViewModel
        registerViewModel.viewStateLiveData.observe(this) { viewState ->
            (viewState as? RegisterViewState)?.let {
                onChangedRegisterViewState(
                    viewState
                )
            }
        }
    }

    private fun onChangedRegisterViewState(viewState: RegisterViewState) {
        when (viewState) {

            is RegisterViewState.Error -> {
                showToast(message = viewState.message)
            }

            is RegisterViewState.EnableInput -> {
                with(binding) {
                    inputEmailRegister.isEnabled = viewState.isEnable
                    inputPassRegisterOk.isEnabled = viewState.isEnable
                    inputPassRegister.isEnabled = viewState.isEnable
                }
            }

            is RegisterViewState.RouteHome -> {
                startActivity(Intent(this@RegisterActivity, HomeActivity::class.java).apply {
                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
                })
            }

            is RegisterViewState.ShowProgress -> {
                binding.progressbar.bringToFront()
                binding.progressbar.isVisible = true
            }

            is RegisterViewState.HideProgress -> {
                binding.progressbar.isVisible = false
            }
        }
    }
}